serde_codegen/src/ser.rs

```diff
@@ -253,8 +255,12 @@ fn serialize_struct(
     );
 
     let type_name = name_expr(builder, item_attrs.name());
-    let len = fields.iter()
+
+    let serialized_fields: Vec<_> = fields.iter()
```

Comment:
We could also make a `Peekable` iterator and check if `fields.peek().is_none()` is true, no?